plugins/jobs/web_client/JobStatus.js

```diff
@@ -47,12 +53,20 @@ var JobStatus = {
     registerStatus: function (status) {
         _.each(status, function (info, name) {
             this[name] = info.value;
-            this._map[info.value] = {
+
+            let statusInfo = {
                 text: info.text,
                 icon: info.icon,
                 color: info.color,
+                textColor: 'white',
                 finished: _.has(info, 'finished') ? info.finished : false
             };
+
+            if ('textColor' in info) {
```

Comment:
I think it goes like this:

If you intend to check up the entire prototype chain to see if the object has any property with the given name, use `in`.

If you only want to check if the object itself has the given property (which is likely what you want here) you should use `hasOwnProperty`, which is equivalent to `_.has`.